app/controllers/page_controller.rb

```diff
@@ -4,18 +4,8 @@ def show
     @page = Page.includes(:page_group).find_by(slug: page_slug.downcase, page_groups: {slug: params[:page_group_friendly_id].downcase})
     @path_parts = request.path.split('/')
     @facilities_data = VaFacility.cached_va_facilities
-    is_admin
-  end
-
-  def is_admin
-    if current_user
-      if not @page.published
-        if not current_user.has_role?(:admin)
-          redirect_to(root_path)
-        end
-      end
-    else
-      redirect_to(root_path)
+    unless @page.published
+      redirect_to(root_path) unless current_user.has_role?(:admin)
```

Comment:
we should do `unless current_user.present? && current_user.has_role?(:admin)` otherwise a non-logged in user that tries to go to an unpublished page will error

Reply:
Ahhh, good catch. I'll fix that up! Thank you!